src/FakeClient/FakeClientMain.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/flags/flag.h>
+#include <absl/flags/parse.h>
+#include <absl/flags/usage.h>
+#include <absl/time/clock.h>
+#include <grpcpp/grpcpp.h>
+
+#include <atomic>
+#include <csignal>
+#include <cstdlib>
+#include <limits>
+#include <memory>
+#include <thread>
+
+#include "FakeCaptureEventProcessor.h"
+#include "GrpcProtos/Constants.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadPool.h"
+#include "OrbitCaptureClient/CaptureClient.h"
+#include "OrbitCaptureClient/CaptureListener.h"
+#include "OrbitClientData/ModuleManager.h"
+#include "capture_data.pb.h"
+
+ABSL_FLAG(uint64_t, port, 44765, "Port OrbitService's gRPC service is listening on");
+ABSL_FLAG(int32_t, pid, 0, "PID of the process to capture");
+ABSL_FLAG(uint32_t, duration, std::numeric_limits<uint32_t>::max(),
+          "Duration of the capture in seconds (stop earlier with Ctrl+C)");
+ABSL_FLAG(uint16_t, sampling_rate, 1000,
+          "Callstack sampling rate in samples per second (0: no sampling)");
+ABSL_FLAG(bool, frame_pointers, false, "Use frame-pointer unwinding");
+ABSL_FLAG(bool, scheduling, true, "Collect scheduling information");
+ABSL_FLAG(bool, thread_state, false, "Collect thread state information");
+ABSL_FLAG(uint16_t, memory_sampling_rate, 0,
+          "Memory usage sampling rate in samples per second (0: no sampling)");
+
+namespace {
+std::atomic<bool> exit_requested = false;
+
+void SigintHandler(int signum) {
+  if (signum == SIGINT) {
+    exit_requested = true;
+  }
+}
+
+// Use SIGINT to stop capturing before the specified duration has elapsed.
+void InstallSigintHandler() {
+  struct sigaction act {};
+  act.sa_handler = SigintHandler;
+  sigemptyset(&act.sa_mask);
+  act.sa_flags = 0;
+  act.sa_restorer = nullptr;
+  sigaction(SIGINT, &act, nullptr);
+}
+
+}  // namespace
+
+// OrbitFakeClient is a simple command line client that connects to a local instance of
+// OrbitService and asks it to take a capture, with capture options specified through command line
+// arguments.
+// It provides a simple way to make OrbitService take a capture in order to tests its performance
+// with various capture options.
+// All the received events are discarded, except for counting them and their total size (see
+// FakeCaptureEventProcessor).
+int main(int argc, char* argv[]) {
+  absl::SetProgramUsageMessage("Orbit fake client for testing");
+  absl::ParseCommandLine(argc, argv);
+
+  FAIL_IF(absl::GetFlag(FLAGS_pid) == 0, "PID to capture not specified");
+  FAIL_IF(absl::GetFlag(FLAGS_duration) == 0, "Specified a zero-length duration");
+
+  int32_t process_id = absl::GetFlag(FLAGS_pid);
+  LOG("process_id=%d", process_id);
+  uint16_t samples_per_second = absl::GetFlag(FLAGS_sampling_rate);
```

Comment:
nit: maybe rename flag to samples_per_second as well?

Reply:
The name of the variable is to match the name of the parameter, but I prefer sampling rate. That's also how this flag is called in other places.